client/src/components/ElementEditor/AddNewButton.js

```diff
@@ -1,14 +1,98 @@
 import React, { Component, PropTypes } from 'react';
+import { InputGroup, Input, InputGroupAddon, Button } from 'reactstrap';
+import i18n from 'i18n';
 
 class AddNewButton extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleTypeChange = this.handleTypeChange.bind(this);
+
+    this.state = {
+      selectedType: null,
+    };
+  }
+
+  /**
+   * Set the current state to the new selected type
+   * @param event
+   */
+  handleTypeChange(event) {
+    const type = event.target
+      ? this.props.elementTypes.find(
+          candidateType => candidateType.value === event.target.value
+        ) || null
+      : null;
+
+    this.setState(() => ({
+      selectedType: type,
+    }));
+  }
+
+  /**
+   * Render the add button next to the dropdown for block types
+   * @returns {DOMElement}
+   */
+  renderAddButton() {
+    const { selectedType } = this.state;
+    const buttonHref = selectedType ? `${this.props.baseAddHref}/${selectedType.value}` : '#';
+    const title = selectedType
+      ? i18n.inject(i18n._t('AddNewButton.TITLE', 'Add a "{type}" block'), { type: selectedType.title })
+      : '';
+    const buttonAttributes = {
+      href: buttonHref,
+      title,
+      disabled: !selectedType,
+      tag: 'a',
+      color: 'primary',
+      className: 'font-icon-plus',
+    };
+
+    return (
+      <Button {...buttonAttributes}>
+        {i18n._t('AddNewButton.ADD', 'Add')}
+      </Button>
+    );
+  }
+
+  /**
+   * Render elemental block type options for a select
+   * @returns {DOMElement}
+   */
+  renderOptions() {
+    const { elementTypes } = this.props;
+
+    if (!elementTypes) {
+      return null;
+    }
+
+    return elementTypes.map(type => (
+      <option value={type.value}>{type.title}</option>
```

Comment:
I think they need a `key` as well - does React not complain about this when compiling? If not, don't worry

Reply:
I didn't get any linting issues or any warnings on compile but I'll recheck when I rebase.